text/0000-natvis.md

```diff
@@ -0,0 +1,491 @@
+extension and specify this Natvis file in the `Cargo.toml` under a new section
+that will be added as part of this RFC.
+
+As an example, consider a crate `foo` with this directory structure:
+
+```text
+/Cargo.toml
+/Foo.natvis (Note: the .natvis file does not have to match the name of the crate.)
+  +-- src
+      +-- main.rs
+```
+
+Where `main.rs` contains:
+
+```rust
+/// A rectangle in first quadrant
+struct FancyRect {
+    pub x: f32,
+    pub y: f32,
+    pub dx: f32,
+    pub dy: f32,
+}
+
+fn main() {
+    let mut fancy_rect = FancyRect::new(10.0, 10.0, 5.0, 5.0);
+    println!("FancyRect: {:?}", fancy_rect);
+}
+```
+
+and `Foo.natvis` contains:
+
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<AutoVisualizer xmlns="http://schemas.microsoft.com/vstudio/debugger/natvis/2010">
+    <Type Name="foo::FancyRect">
+      <DisplayString>({x},{y}) + ({dx}, {dy})</DisplayString>
+      <Expand>
+        <Synthetic Name="LowerLeft">
+          <DisplayString>({x}, {y})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperLeft">
+          <DisplayString>({x}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="UpperRight">
+          <DisplayString>({x + dx}, {y + dy})</DisplayString>
+        </Synthetic>
+        <Synthetic Name="LowerRight">
+          <DisplayString>({x + dx}, {y})</DisplayString>
+        </Synthetic>
+      </Expand>
+    </Type>
+</AutoVisualizer>
+```
+
+When viewed under WinDbg, the `fancy_rect` variable would be shown as follows:
+
+```text
+> Variables:
+  > fancy_rect: (10, 10) + (5, 5)
+    > LowerLeft: (10, 10)
+    > UpperLeft: (10, 15)
+    > UpperRight: (15, 15)
+    > LowerRight: (15, 10)
+```
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+In rustc, a new `-C natvis={comma-separated list of .natvis files}` flag
+will be added which instructs the compiler to take the set of .natvis
+files for a given crate and store them in the crate metadata. This will be gated
+by the existing `-Z unstable-options` flag to ensure it is only used in unstable
+Rust. When running the linker, using the `MSVC` toolchain, a `/NATVIS` linker
+option would be set for each `.natvis` file. This includes `.natvis` files from
+all crate dependencies, if any exist, as well as the current crate and embed them
+into the PDB.
+
+The MSVC linker supports embedding debugger visualizations defined in a Natvis file
+(`.natvis`) into a PDB generated by LINK through the use of the `/NATVIS` linker flag.
+Since the `MSVC` linker is the only one that supports embedding Natvis files
+into a PDB, this feature would be specific to the `MSVC` toolchain only.
+
+Cargo would add new syntax in `Cargo.toml` for specifying the list of Natvis
+files to be added to the crate. The new manifest key, `natvis` would be added
+to a new `[debug-visualizations]` section. This would be in control of setting
```

Comment:
```suggestion
to a new `[debugger-visualizations]` section. This would be in control of setting
```

Maybe? Seems like there might be confusion with the Debug trait or debug profile, etc.

Reply:
I'll update this to `debugger-visualizations`.